src/SourceWatcherFuncApp/DiagSourceWatcher.cs

```diff
@@ -48,10 +54,14 @@ public async Task Run([TimerTrigger("0 0 0 * * *")]TimerInfo timerInfo, ILogger
 
                 foreach (var githubdir in githubDirectories)
                 {
-                    if (!githubdir.Type.Equals("dir", StringComparison.OrdinalIgnoreCase)) continue;
+                    if (!githubdir.Type.Equals("tree", StringComparison.OrdinalIgnoreCase)) continue;
 
                     try
                     {
+                        if(githubdir.Url.Contains("trees"))
+                        {
+                            githubdir.Url = $"https://api.github.com/repos/{config["Github:UserName"]}/{config["Github:RepoName"]}/contents/{githubdir.Name}?ref={config["Github:Branch"]}";
```

Comment:
I added a new method to the `IGitHubService` that does this if you'll like to use it. https://github.com/Azure/Azure-AppServices-Diagnostics/blob/main/src/SourceWatcherFuncApp/Services/IGithubService.cs#L190-L193